File: php/slim-doctrine/app/src/com/sprint/sms/api/access/ApiAccess.php
<?php
namespace App\com\sprint\sms\api\access;

use \App\com\sprint\sms\api\support\FormatSupport;
use \App\com\sprint\sms\api\util\AppConstant;

class ApiAccess implements FormatSupport
{
	private $description;
	
	private $parameter = array();
	
	public function getDescription() {
		return $this->description;
	}

	public function setDescription($description) {
		$this->description = $description;
	}

	public function getParameter() {
		return $this->parameter;
	}

	public function setParameter(array $parameter) {
		$this->parameter = $parameter;
	}
	
	public function toFormatObject() {
		$o = array();
		$o[AppConstant::API_KEY_DESCRIPTION] = $this->description;	
		if ($this->parameter !== null && count($this->parameter) !== 0) {
			$o[AppConstant::API_KEY_PARAMETER] = $this->parameter;
		}
		return $o;
	}
	
}

File: java/spring-hibernate/src/main/java/com/sprint/sms/api/access/ApiAccess.java
package com.sprint.sms.api.access;

import java.io.Serializable;
import java.util.HashMap;
import java.util.Map;

import com.ideahut.common2.annotation.IdhFormatter;

@IdhFormatter
public class ApiAccess implements Serializable {

	/**
	 * 
	 */
	private static final long serialVersionUID = 4639659282867599284L;

	private String description;
	
	private Map<String, String> parameter = new HashMap<String, String>();

	@IdhFormatter
	public String getDescription() {
		return description;
	}

	public void setDescription(String description) {
		this.description = description;
	}

	@IdhFormatter
	public Map<String, String> getParameter() {
		return parameter;
	}

	public void setParameter(Map<String, String> parameter) {
		this.parameter = parameter;
	}
	
}
